client/client.go

```diff
@@ -394,6 +420,29 @@ func (c *Client) Node() *structs.Node {
 	return c.config.Node
 }
 
+// StatsReporter exposes the various APIs related resource usage of a Nomad
+// client
+func (c *Client) StatsReporter() ClientStatsReporter {
+	return c
+}
+
+// AllocStats returns all the stats reporter of the allocations running on a
+// Nomad client
+func (c *Client) AllocStats() map[string]AllocStatsReporter {
+	res := make(map[string]AllocStatsReporter)
```

Comment:
@dadgar I think I should use the `getAllocRunners` method, which locks and returns a snapshot of the alloc runners.